src/main/java/edu/unc/lib/boxc/migration/cdm/services/GroupMappingService.java

```diff
@@ -109,58 +105,28 @@ public void generateMapping(GroupMappingOptions options) throws Exception {
-        ResultSet groupRs = stmt.executeQuery("select " + options.getGroupField().get(0)
-                + " from " + CdmIndexService.TB_NAME
-                + " where " + CdmIndexService.ENTRY_TYPE_FIELD + " is null"
-                + " group by " + options.getGroupField().get(0)
-                + " having count(*) > 1");
-        while (groupRs.next()) {
-            var groupValue = groupRs.getString(1);
-            if (StringUtils.isBlank(groupValue)) {
-                continue;
-            }
-            multiMemberGroupSet.add(groupValue);
-        }
-
-        ResultSet rs = stmt.executeQuery("select " + CdmFieldInfo.CDM_ID + ", " + options.getGroupField().get(0)
-                + " from " + CdmIndexService.TB_NAME
-                + " where " + CdmIndexService.ENTRY_TYPE_FIELD + " is null"
-                + " order by " + CdmFieldInfo.CDM_ID + " ASC");
-        while (rs.next()) {
-            String cdmId = rs.getString(1);
-            String matchedValue = rs.getString(2);
-
-            // Add empty mapping for records either not in groups or in groups with fewer than 2 members
-            if (StringUtils.isBlank(matchedValue) || !multiMemberGroupSet.contains(matchedValue)) {
-                log.debug("No matching field for object {}", cdmId);
-                csvPrinter.printRecord(cdmId, null);
-                continue;
-            }
-
-            String groupKey = GroupMappingInfo.GROUPED_WORK_PREFIX + options.getGroupField().get(0) + ":" + matchedValue;
-            csvPrinter.printRecord(cdmId, groupKey);
-        }
-    }
-
     private void generateMultipleGroupMapping(GroupMappingOptions options, Statement stmt,
-                                              HashSet<String> multiMemberGroupSet, CSVPrinter csvPrinter) throws Exception {
-        int numberGroups = options.getGroupField().size();
-        String multipleGroups = String.join(", ", options.getGroupField());
+                                              Set<String> multiMemberGroupSet, CSVPrinter csvPrinter) throws Exception {
+        int numberGroups = options.getGroupFields().size();
+        String multipleGroups = String.join(", ", options.getGroupFields());
 
         ResultSet groupRs = stmt.executeQuery("select " + multipleGroups
                 + " from " + CdmIndexService.TB_NAME
                 + " where " + CdmIndexService.ENTRY_TYPE_FIELD + " is null"
                 + " group by " + multipleGroups
                 + " having count(*) > 1");
         while (groupRs.next()) {
+            List<String> groupValues = new ArrayList<>();
             for (int i = 1; i < numberGroups + 1; i++) {
                 var groupValue = groupRs.getString(i);
                 if (StringUtils.isBlank(groupValue)) {
                     continue;
                 }
-                multiMemberGroupSet.add(groupValue);
+                groupValues.add(groupValue);
+            }
+            if (!groupValues.isEmpty()) {
```

Comment:
This is good, but one case we're still not covering is key collisions with blank fields. For example, currently these two records would produce the same key, even though their data is different:
```
cdmid: 1
field1: a
field2:

cdmid: 2
field1:
field2: a
```
They'd both end up with "a" as the key. If we leave blanks in the key, we'd get "a," and ",a" as the keys, so they wouldn't get grouped together incorrectly. 

So I'd suggest allowing blanks into `groupValues` and `matchedValues`, and then adding a helper method for building the key which take a list of values and would:
1. Return null or empty string if every value in the list is empty
2. Otherwise, join all the values like you're currently doing